### PR TITLE
Add docs for `Container`

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description

I've considered the [old PR](https://github.com/mckinsey/vizro/pull/206) on this, but made some changes:
- Created an initial overview page for components (we had several places where it became clear that we needed a landing page for components as otherwise it was referencing the Card & Button docs)
- Changed the example for container 
- Updated docs on it

Will be merged after https://github.com/mckinsey/vizro/pull/254

## Screenshot
The overview page can definitely be improved further, but I wouldn't do that in this PR. So currently it's just a basic version.
Screenshot 2024-01-11 at 16 16 50

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/docs/pages/user_guides/card_button.md [hunk 1]
```diff
@@ -0,0 +1,703 @@
+# How to use cards & buttons
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user_guides/card_button.md:1]: Nothing new, I've just renamed the file from `components.md` to `card_button.md`

File: vizro-core/docs/pages/user_guides/graph.md [hunk 2]
```diff
@@ -2,9 +2,6 @@
 
 This guide shows you how to use graphs to visualize your data in the dashboard.
 
-The [`Page`][vizro.models.Page] models accepts the `components` argument, where you can enter your visual content e.g.
-[`Graph`][vizro.models.Graph], [`Table`][vizro.models.Table], [`Card`][vizro.models.Card] or [`Button`][vizro.models.Button].
-
 ## Graph
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user_guides/graph.md:6]:
```suggestion
```
Would remove now completely

File: vizro-core/docs/pages/user_guides/table.md [hunk 3]
```diff
@@ -2,9 +2,6 @@
 
 This guide shows you how to use tables to visualize your data in the dashboard.
 
-The [`Page`][vizro.models.Page] models accepts the `components` argument, where you can enter your visual content e.g.
-[`Graph`][vizro.models.Graph], [`Table`][vizro.models.Table], [`Card`][vizro.models.Card] or [`Button`][vizro.models.Button].
-
 ## Table
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user_guides/table.md:6]:
```suggestion
```
Similar here

File: vizro-core/docs/pages/user_guides/pages.md [hunk 4]
```diff
@@ -22,7 +22,7 @@ A [`Page`][vizro.models.Page] is split up into four main containers:
 To create and add a page to your dashboard, do the following steps:
 
 1. Provide a `title` to your [`Page`][vizro.models.Page]
-2. Configure your `components`, see our guide on [Charts/Components](components.md)
+2. Configure your `components`, see our guide on the various options [here](components.md).
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user_guides/pages.md:25]:
```suggestion
2. Configure your `components`, see our guide on the [various options](components.md).
```

File: vizro-core/docs/pages/user_guides/components.md
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user_guides/components.md:1]: Love the overview page, I think it makes a lot of sense!!

Do not love the design of the cards too much though, the dark theme bites quite a bit. Two ideas to improve:
- use light theme (J might not be happy ;) )
- just remove the screenshots and work with nothing or icons? I liked for instance some of the stuff here: https://squidfunk.github.io/mkdocs-material/reference/grids/

File: vizro-core/docs/pages/user_guides/container.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
+# How to use containers
+
+This guide shows you how to use containers to organize your page components into subsections inside the dashboard.
+
+## Container
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user_guides/container.md:5]:
```suggestion
```

**Review by maxschulz-COL — APPROVED** [review]
Lgtm! Thanks for writing this so quickly. Have a few suggestions that might make it even slicker

File: vizro-core/docs/pages/user_guides/components.md
Comment (huong-li-nguyen) [reply vizro-core/docs/pages/user_guides/components.md:1]: Yes, I agree, but I just wanted to get a simple overview page out. There should be a separate PR on optimizing this, so in the meantime I can replace the screenshots to white theme or just take them out completely and add some text instead.

Comment (huong-li-nguyen) [reply vizro-core/docs/pages/user_guides/components.md:1]: Screenshot 2024-01-11 at 16 05 00

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -6,6 +6,7 @@ repos:
       - id: check-merge-conflict
       - id: check-case-conflict
       - id: check-yaml
+        args: ["--unsafe"]
```
Comment (huong-li-nguyen) [inline .pre-commit-config.yaml:9]: Required such that it only checks for syntax and does not try to evaluate the yaml: https://stackoverflow.com/questions/59413979/how-exclude-ref-tag-from-check-yaml-git-hook

For the evaluation we have Alexeys tests anyways. Without this, linting will not pass as it does not understand the code below in the mkdocs.yml, which we need for the icons.

```
  - pymdownx.emoji:
      emoji_index: !!python/name:material.extensions.emoji.twemoji
      emoji_generator: !!python/name:material.extensions.emoji.to_svg
```

**Review by nadijagraca — APPROVED** [review]
LGTM! I really like the overview page. 🚀

File: vizro-core/docs/pages/user_guides/components.md
Comment (maxschulz-COL) [reply vizro-core/docs/pages/user_guides/components.md:1]: I like this a lot more ❤️

**Review by antonymilne — APPROVED** [review]
Thank you for doing this! ⭐ And sorry it took so long to review, I know it shouldn't have. Normally I don't bother looking much at docs changes but this is one case I wanted to see if it conveyed the "essence" of what I think a container represents.

I think this is great as a sort of MVP docs page but could do with some more detailed information, e.g.
* more context around what is the point of a container and how `Container` is almost the same as `Page` so that users don't feel like they're learning a whole new paradigm
* why would anyone want to use a `Container` given you can do something very similar using `page.layout`? 
* how you can nest containers and how `page.layout` and `container.layout` use the same system so that you can achieve "finer grained" control of your grid. Maybe an example of nested containers

To bring it to life, maybe think of something like this example:
```
vm.Page(layout=[[0, 1], [2, 2]], components=[vm.Card(id=0), vm.Card(id=1), vm.Card(id=2)])
```
compared to
```
vm.Page(layout=[[0], [1]], components=[Container(layout=[0, 1], components=[vm.Card(id=0), vm.Card(id=1)), vm.Card(id=2)])
```

These render very similarly on the screen. What's the difference between them? (My answer: at the moment there's no difference apart from `Container` must have a title, but in future it will have `controls` also that control just things in that container. A container is a way to group related components together into a section on the page.)

Up to you to decide which of these suggestions, if any, to incorporate now, and which to leave for future changes though - I certainly don't want to delay this merging any more if you think we shouldn't.

File: vizro-core/docs/pages/user_guides/container.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+This guide shows you how to use containers to organize your page components into subsections inside the dashboard.
+
+To add a [`Container`][vizro.models.Container] to your page, do the following:
+
+1. Insert the [`Container`][vizro.models.Container] into the `components` argument of the [`Page`][vizro.models.Page]
+2. Provide a `title` to your [`Container`][vizro.models.Container]
+3. Configure your `components`, see our overview page on the various options [here](components.md)
+4. (optional) Configure your `layout` , see our guide on [Layouts](layouts.md)
+
+!!! example "Container"
```
[the remark this answers is not in the document]
Comment (huong-li-nguyen) [reply vizro-core/docs/pages/user_guides/container.md:12]: Thanks! Fixed it - it was because of a space 😅

**Comment by huong-li-nguyen** [comment]
This makes a lot of sense! Let me try to rephrase some of this in a separate PR.
 
I've added the quick changes already. The rest I'm currently pushing to a separate PR. So there is no time rush of getting these in right now, but if we manage then that's great! 👍

**Pull request merged**